jupyterhub/templates/hub/deployment.yaml

```diff
@@ -69,9 +74,59 @@ spec:
         {{- end }}
         {{- end }}
       {{- end }}
-      {{- if .Values.hub.initContainers }}
+      {{- if or .Values.hub.initContainers .Values.hub.https.enabled }}
       initContainers:
+        {{- if .Values.hub.https.enabled }}
+        - name: init-tls
+          image: {{ .Values.hub.image.name }}:{{ .Values.hub.image.tag }}
+          command:
+            - "/bin/sh"
+            - "-ec"
+            - |
+              jupyterhub -f /etc/jupyterhub/jupyterhub_config.py --generate-certs
+              cd /srv/jupyterhub-internal-ssl/internal-ssl/
+              kubectl create secret generic \
+                --namespace $POD_NAMESPACE \
+                jupyterhub-certs \
+                --from-file=certipy.json \
+                --from-file=services-ca/services-ca.key \
+                --from-file=services-ca/services-ca.crt \
+                --from-file=notebooks-ca/notebooks-ca.key \
+                --from-file=notebooks-ca/notebooks-ca.crt \
+                --from-file=proxy-api-ca/proxy-api-ca.key \
+                --from-file=proxy-api-ca/proxy-api-ca.crt \
+                --from-file=proxy-client-ca/proxy-client-ca.key \
+                --from-file=proxy-client-ca/proxy-client-ca.crt \
+                --from-file=hub-ca/hub-ca.key \
+                --from-file=hub-ca/hub-ca.crt \
+                --from-file=hub-ca_trust.crt \
+                --from-file=proxy-api-ca_trust.crt \
+                --from-file=proxy-client-ca_trust.crt \
+                --from-file=notebooks-ca_trust.crt \
+                --from-file=services-ca_trust.crt \
+                --from-file=hub-internal/hub-internal.key \
+                --from-file=hub-internal/hub-internal.crt \
+                --from-file=proxy-api/proxy-api.key \
+                --from-file=proxy-api/proxy-api.crt \
+                --from-file=proxy-client/proxy-client.key \
+                --from-file=proxy-client/proxy-client.crt \
+                --dry-run -o yaml > jupyterhub-certs-secret.yaml
+                kubectl --namespace $POD_NAMESPACE create -f jupyterhub-certs-secret.yaml || kubectl --namespace $POD_NAMESPACE replace -f jupyterhub-certs-secret.yaml
```

Comment:
I can't recall myself. I think it only re-generates them if they're missing or if they're expired or something.

I also believe there's more information in the secret than needed, but if I recall, it was still necessary for some reason.

Reply:
If nothing else, it was probably just a short-cut to make it work, rather than make it perfect.

Reply:
It may make sense to convert this initContainer to mount and execute some sort of python script that imports `kubernetes`, since it's already included in the hub container. 

This would allow for only having to maintain support for the kubernetes api version in one location.